pkg/destroy/azure/azure.go

```diff
@@ -419,17 +423,57 @@ func deleteAzureStackPublicRecords(ctx context.Context, o *ClusterUninstaller) e
 	return utilerrors.NewAggregate(errs)
 }
 
-func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, recordsClient dns.RecordSetsClient, privateDNSClient privatedns.PrivateZonesClient, privateRecordsClient privatedns.RecordSetsClient, logger logrus.FieldLogger, rgName string) error {
+func deleteRecordsFromBaseDomain(ctx context.Context, o *ClusterUninstaller) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
+
+	var errs []error
+	result, err := o.recordsClient.Get(ctx, o.BaseDomainResourceGroupName, o.ZoneName, o.APIURL, dns.CNAME)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to get base domain dns zone: %w", err))
+		if isAuthError(err) {
+			return err
+		}
+	}
+
+	if value, ok := result.Metadata["InfraID"]; ok {
+		if value == &o.InfraID {
+			deleteResult, err := o.recordsClient.Delete(ctx, o.BaseDomainResourceGroupName, o.ZoneName, o.APIURL, dns.CNAME, "")
+			if err != nil {
+				if deleteResult.IsHTTPStatus(http.StatusNotFound) {
+					o.Logger.Debug("already deleted")
+					err2 := deleteRecordsFromBaseDomain(ctx, o)
+					if err2 != nil {
+						o.Logger.Debugf("failed to delete record sets from the base domain: %w", err)
+					}
+					return utilerrors.NewAggregate(errs)
+				}
+				errs = append(errs, fmt.Errorf("failed to delete base domain dns zone: %w", err))
+				if isAuthError(err) {
+					return err
+				}
+			}
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func deletePublicRecords(ctx context.Context, o *ClusterUninstaller) error {
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
 	defer cancel()
 
 	// collect records from private zones in rgName
 	var errs []error
 
-	zonesPage, err := dnsClient.ListByResourceGroup(ctx, rgName, to.Int32Ptr(100))
+	zonesPage, err := o.zonesClient.ListByResourceGroup(ctx, o.ResourceGroupName, to.Int32Ptr(100))
 	if err != nil {
 		if zonesPage.Response().IsHTTPStatus(http.StatusNotFound) {
-			logger.Debug("already deleted")
+			o.Logger.Debug("already deleted")
+			err2 := deleteRecordsFromBaseDomain(ctx, o)
```

Comment:
perhaps make running this conditional on the presence of `zoneName` in the metadata, or you could put the `zoneName` check in this function and return nil or something if `zoneName` is missing.

Reply:
There's an error check in the function for the 3 inputs and it returns without doing anything if any of the 3 is empty.